docs/pyproject.toml

```diff
@@ -9,7 +9,7 @@ python = "^3.7"
 pyyaml = "5.3"
 pygments = "2.2.0"
 recommonmark = "0.5.0"
-sphinx-scylladb-theme = "~0.1.11"
+sphinx-scylladb-theme = "~1.0.0"
```

Comment:
I think here (and everywhere else) we should pin with equality for the reasons mentioned above

Reply:
"~" means update only if there is a new patch.

This allows us to release new theme versions with nonbreaking changes without having to submit a PR per repository.

If the major or minor is updated, the theme is not updated automatically.

More information: https://github.com/scylladb/sphinx-scylladb-theme/issues/75

Reply:
I still believe we need to pin the dependencies but we could leave this and just add the poetry-lock file which solve this.

One of the goals of the CI run is reproducibility which would be negated by running with different dependency versions.

We needed to fix the python deps and while adding CI for it I had to pick most of the pieces from this PR in #886